ion/symboltoken_test.go

```diff
@@ -0,0 +1,236 @@
+	text1   *string
+	sid1    int64
+	source1 *ImportSource
+	text2   *string
+	sid2    int64
+	source2 *ImportSource
+}{
+	{nil, 123, nil,
+		nil, 123, nil},
+	{nil, 123, newSource("table", 1),
+		nil, 123, newSource("table", 1)},
+	{newString("text1"), 123, nil,
+		newString("text1"), 123, nil},
+	{newString("text2"), 123, newSource("table", 1),
+		newString("text2"), 123, newSource("table", 1)},
+}
+
+func TestBoolEqualsOperator(t *testing.T) {
+	for _, testData := range boolEqualsTestData {
+		st1 := SymbolToken{Text: testData.text1, LocalSID: testData.sid1, Source: testData.source1}
+		st2 := SymbolToken{Text: testData.text2, LocalSID: testData.sid2, Source: testData.source2}
+
+		if !st1.Equal(&st2) {
+			t.Errorf("expected %v, got %v", true, false)
+		}
+	}
+}
+
+var boolNotEqualsTestData = []struct {
+	text1   *string
+	sid1    int64
+	source1 *ImportSource
+	text2   *string
+	sid2    int64
+	source2 *ImportSource
+}{
+	{nil, 123, newSource("table", 1),
+		nil, 123, nil},
+	{nil, 123, nil,
+		newString("text1"), 123, nil},
+	{nil, 123, newSource("table", 1),
+		nil, 123, newSource("table2", 1)},
+	{nil, 123, newSource("table", 1),
+		nil, 123, newSource("table", 2)},
+	{newString("text2"), 123, nil,
+		newString("text3"), 123, nil},
+}
+
+func TestBoolNotEqualsOperator(t *testing.T) {
+	for _, testData := range boolNotEqualsTestData {
+		st1 := SymbolToken{Text: testData.text1, LocalSID: testData.sid1, Source: testData.source1}
+		st2 := SymbolToken{Text: testData.text2, LocalSID: testData.sid2, Source: testData.source2}
+
+		if st1.Equal(&st2) {
+			t.Errorf("expected %v, got %v", false, true)
+		}
+	}
+}
+
+// Make sure SymbolToken conforms to Stringer
+var _ fmt.Stringer = &SymbolToken{}
+
+func TestSymbolToken_String(t *testing.T) {
+	cases := []struct {
+		desc     string
+		token    SymbolToken
+		expected string
+	}{
+		{
+			desc: "Text and SID",
+			token: SymbolToken{
+				Text:     newString("hello"),
+				LocalSID: 10,
+				Source:   nil,
+			},
+			expected: `{"hello" 10 nil}`,
+		},
+		{
+			desc: "nil Text",
+			token: SymbolToken{
+				Text:     nil,
+				LocalSID: 11,
+				Source:   nil,
+			},
+			expected: `{nil 11 nil}`,
+		},
+		{
+			desc: "Text and SID with Import",
+			token: SymbolToken{
+				Text:     newString("world"),
+				LocalSID: 12,
+				Source:   newSource("foobar", 3),
+			},
+			expected: `{"world" 12 {"foobar" 3}}`,
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.desc, func(t *testing.T) {
+			if diff := cmp.Diff(c.expected, c.token.String()); diff != "" {
+				t.Errorf("Token String() differs (-expected, +actual):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestNewImportSource(t *testing.T) {
+	is := newSource("table", 1)
+	if is.Table != "table" {
+		t.Errorf("expected %v, got %v", "table", is.Table)
+	}
+	if is.SID != 1 {
+		t.Errorf("expected %v, got %v", 1, is.SID)
+	}
+}
+
+var ImportSourceBoolEqualsTestData = []struct {
+	text1 string
+	sid1  int64
+	text2 string
+	sid2  int64
+}{
```

Comment:
If we add an extra element here for the expected value of test (true or false) then the two sets of tests can be collapsed to 1 set of tests.

Reply:
Updated.